File: Python/Easy/palindromeNumber.py
class Solution(object):
    def isPalindromeStr(self, x):
        """
        :type x: int
        :rtype: bool
        """
        return str(x) == str(x)[::-1]
    
    def isPalindromeInt(self, x):
        original_x = x
        reversed_x = 0
        if x < 0:
            return False
        while x > 0:
            digit = x % 10
            reversed_x = reversed_x * 10 + digit
            x //= 10
            
        return original_x == reversed_x
    
def main():
    solution = Solution()
    num1 = 121
    num2 = 125
    print(solution.isPalindromeStr(num1))
    print(solution.isPalindromeStr(num2))
    print(solution.isPalindromeInt(num1))
    print(solution.isPalindromeInt(num2))

if __name__ == "__main__":
    main()

File: JS/Easy/palindromeNumber.js
class Solution {
    isPalindromeStr(x) {
        const str = x.toString();
        return str === [...str].reverse().join('');
    }

    isPalindromeInt(x) {
        if (x < 0) return false;

        let originalX = x;
        let reversedX = 0;

        while (x > 0) {
            const digit = x % 10;
            reversedX = reversedX * 10 + digit;
            x = Math.floor(x / 10);
        }

        return originalX === reversedX
    }
}

function main() {
    const solution = new Solution();
    const num1 = 121;
    const num2 = 125;
    
    console.log(solution.isPalindromeStr(num1));
    console.log(solution.isPalindromeStr(num2));

    console.log(solution.isPalindromeInt(num1));
    console.log(solution.isPalindromeInt(num2));
}

main();